src/js/components/agencyV2/visualizations/totalObligationsOverTime/AgencyBudgetLine.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+const propTypes = {
+    data: PropTypes.array,
+    xScale: PropTypes.func,
+    xDomain: PropTypes.array,
+    yScale: PropTypes.func,
+    height: PropTypes.number,
+    width: PropTypes.number,
+    agencyBudget: PropTypes.number,
+    todaysDate: PropTypes.number,
+    padding: PropTypes.object
+};
+
+const AgencyBudgetLine = ({
+    data,
+    xScale,
+    xDomain,
+    yScale,
+    height,
+    agencyBudget,
+    todaysDate,
+    padding,
+    width
+}) => {
+    const [show, setShow] = useState(false);
+    const [lineData, setLineData] = useState({
+        x1: 0,
+        x2: 0,
+        y1: 0
+    });
+    const [rectangleData, setRectangleData] = useState({
+        x: 0,
+        y: 0,
+        width: 0,
+        height: 0
+    });
+
+    useEffect(() => {
+        if ((todaysDate >= xDomain[0]) && (todaysDate <= xDomain[1])) {
+            setShow(true);
+        }
+        else {
+            setShow(false);
+        }
+    }, [xScale, xDomain, todaysDate]);
+
+    useEffect(() => {
+        if (xScale && yScale && agencyBudget) {
+            setLineData(
+                {
+                    x1: padding.left,
+                    x2: show ? xScale(todaysDate) + padding.left : width - padding.left,
+                    y1: height - yScale(agencyBudget) - padding.top
```

Comment:
can migrate this `y1` to it's own variable and check if it is `isNaN` to remove the console errors?

```
const y = height - yScale(agencyBudget) - padding.top;
```

then in the object

```
y1: isNaN(y1) ? 0 : y1
```

Reply:
done!